Error with array of generic structs

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->
**V doctor:**
```
OS: linux, "Arch Linux"
Processor: 8 cpus, 64bit, little endian, Intel(R) Core(TM) i5-1035G1 CPU @ 1.00GHz
CC version: cc (GCC) 11.1.0

getwd: /home/ulisesjcf/Projects/Github/vtl
vmodules: /home/ulisesjcf/.vmodules
vroot: /home/ulisesjcf/Projects/Github/v
vexe: /home/ulisesjcf/Projects/Github/v/v
vexe mtime: 2021-10-13 03:40:17
is vroot writable: true
is vmodules writable: true
V full version: V 0.2.4 3c7c11e.e05095c

Git version: git version 2.33.0
Git vroot status: weekly.2021.41-20-ge05095cf
.git/config present: true
thirdparty/tcc status: thirdparty-linux-amd64 333c14de

```

**What did you do?**
`v -g -o vdbg cmd/v && vdbg main.v`
```v
pub struct Abcd<T> {}
pub fn iterators<T>() []&Abcd<T> { return []&Abcd<T>{} }
iterators<f64>()

```

**What did you expect to see?**

The program works as expected


**What did you see instead?**
```
==================
/tmp/v/main.949355795085655900.tmp.c:1782: error: ';' expected (got "<")
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang

```

--- a/vlib/v/checker/checker.v
+++ b/vlib/v/checker/checker.v
@@ -685,6 +685,20 @@ fn (mut c Checker) unwrap_generic_type(typ ast.Type, generic_names []string, con
 	mut c_nrt := ''
 	ts := c.table.get_type_symbol(typ)
 	match mut ts.info {
+		ast.Array {
+			mut elem_type := ts.info.elem_type
+			mut elem_sym := c.table.get_type_symbol(elem_type)
+			mut dims := 1
+			for mut elem_sym.info is ast.Array {
+				info := elem_sym.info as ast.Array
+				elem_type = info.elem_type
+				elem_sym = c.table.get_type_symbol(elem_type)
+				dims++
+			}
+			unwrap_typ := c.unwrap_generic_type(elem_type, generic_names, concrete_types)
+			idx := c.table.find_or_register_array_with_dims(unwrap_typ, dims)
+			return ast.new_type(idx).derive_add_muls(typ).clear_flag(.generic)
+		}
 		ast.Struct, ast.Interface, ast.SumType {
 			if !ts.info.is_generic {
 				return typ

--- a/vlib/v/tests/generics_fn_return_array_of_generic_struct_test.v
+++ b/vlib/v/tests/generics_fn_return_array_of_generic_struct_test.v
@@ -1,0 +1,12 @@
+pub struct Abcd<T> {
+}
+
+pub fn iterators<T>() []&Abcd<T> {
+	return []&Abcd<T>{}
+}
+
+fn test_generic_fn_return_array_of_generic_struct() {
+	a := iterators<f64>()
+	println(a)
+	assert '$a' == '[]'
+}

